submit-web/src/router.tsx

```diff
@@ -42,6 +47,29 @@ export default function RouterProviderWithAuthContext() {
   const account = useAccount();
   const { setAccount } = account;
 
+  const [termsAccepted, setTermsAccepted] = useState(false);
```

Comment:
this file, router.tsx is supposed to be a router file. Now it's doing so much. Can we revisit this and see where we can put the terms of service logic